src/bsoncxx/builder/core.cpp

```diff
@@ -410,7 +410,11 @@ void core::concatenate(const bsoncxx::document::view& view) {
 }
 
 void core::append(const bsoncxx::types::value& value) {
-    switch (static_cast<int>(value.type())) {
+    if (!value.type()) {
+        throw(std::runtime_error("invalid value"));
```

Comment:
aha, something that belongs in the bsoncxx error category!!  Sweet.  Maybe mark this with a todo about the exception being thrown.